test/python/quantum_info/test_synthesis.py

```diff
@@ -173,6 +174,28 @@ def assertRoundTrip(self, weyl1: TwoQubitWeylDecomposition):
         self.assertEqual(maxdiff, 0, msg=f"K2r matrix differs by {maxdiff}" + msg_base)
         self.assertEqual(weyl1.requested_fidelity, weyl2.requested_fidelity, msg_base)
 
+    def assertRoundTripPickle(self, weyl1: TwoQubitWeylDecomposition):
+        """Fail if loads(dumps(weyl1)) not equal to weyl1"""
+
+        pkl = pickle.dumps(weyl1, protocol=max(4, pickle.DEFAULT_PROTOCOL))
+        weyl2 = pickle.loads(pkl)
+        msg_base = f"weyl1:\n{weyl1}\nweyl2:\n{repr(weyl2)}"
```

Comment:
I learned recently that `f"{repr(x)}"` can also be written as `f"{x!r}"` (and `!s` for `str`), which seems like a fun way to introduce a million more sigils to the strings and make them completely unreadable.